src/Storages/MergeTree/RequestResponse.cpp

```diff
@@ -95,14 +106,21 @@ void PartitionReadRequest::deserialize(ReadBuffer & in)
 
 UInt64 PartitionReadRequest::getConsistentHash(size_t buckets) const
 {
-    auto hash = SipHash();
+    SipHash hash;
+
+    hash.update(partition_id.size());
```

Comment:
What's the difference after we consider size?

Reply:
In the previous version, only the string contents were hashed, so some combinations of strings may yield identical hashes:
('a', 'bc')
('ab', 'c')

It is a non-significant change, as this hash allows to have collisions, it is more like a readability change, so people will not stumble upon this pice of code thinking it is incorrect.

Reply:
it's ok